Frameworks/CoreGraphics/CGPattern.mm

```diff
@@ -233,3 +233,8 @@ CGRect _CGPatternGetFinalPatternSize(CGPatternRef pattern) {
     RETURN_RESULT_IF_NULL(pattern, CGRectNull);
     return { CGPointZero, { ((CGPattern*)pattern)->xStep, ((CGPattern*)pattern)->yStep } };
 }
+
+bool _CGPatternIsColored(CGPatternRef pattern) {
+    FAIL_FAST_IF_NULL(pattern);
```

Comment:
nit: we shouldn't fail fast in private functions. Our pattern has been to return an HRESULT and fail in the base CG API function. That said, I don't think you need the null check in particular here. Since it's a private method, move the fail_fast_if_null to CGContext. If you plan on using this method in more places though I would recommend keeping the fail fast here and returning a struct for {HRESULT, returnvalue} which has been our preferred pattern or passing a return bool by reference.